builder/shell-for.nix

```diff
@@ -127,7 +127,22 @@ in
     nativeBuildInputs = [ ghcEnv ]
       ++ nativeBuildInputs
       ++ mkDrvArgs.nativeBuildInputs or []
-      ++ lib.attrValues (buildPackages.haskell-nix.tools compiler.nix-name tools);
+      ++ lib.attrValues (buildPackages.haskell-nix.tools compiler.nix-name tools)
+      # If this shell is a cross compilation shell include
+      # wrapper script for running cabal build with appropriate args.
+      ++ lib.optional (ghcEnv.targetPrefix != "") (
+            buildPackages.writeShellScriptBin "${ghcEnv.targetPrefix}cabal" ''
+              exec cabal \
+                --with-ghc=${ghcEnv.targetPrefix}ghc \
```

Comment:
Some of these flags appear a few times in a few places - is there a way we could abstract them? For example, could we use the same wrapped derivation for a wrapped cabal script in the component builder, and thus share it with the shell?

Reply:
Sounds like a good idea for maintainability.
The component builder uses bin/Setup instead of cabal, of course.
And there are probably some configure flags that you want for the component builder but not for interactive cabal.
But there should be a common subset of flags that need not be duplicated.